src/hotspot/share/c1/c1_Instruction.hpp

```diff
@@ -2402,12 +2299,23 @@ BASE(UnsafeObjectOp, UnsafeOp)
 
 
 LEAF(UnsafeGetObject, UnsafeObjectOp)
+ private:
+  bool _is_raw_get;
  public:
   UnsafeGetObject(BasicType basic_type, Value object, Value offset, bool is_volatile)
   : UnsafeObjectOp(basic_type, object, offset, false, is_volatile)
   {
     ASSERT_VALUES
+    _is_raw_get = false;
   }
+  UnsafeGetObject(BasicType basic_type, Value object, Value offset, bool is_volatile, bool is_raw_get)
+  : UnsafeObjectOp(basic_type, object, offset, false, is_volatile), _is_raw_get(is_raw_get)
+  {
+    ASSERT_VALUES
+  }
+
+  // accessors
+  bool is_raw_get()                             { return _is_raw_get; }
```

Comment:
I would rename this to `_is_raw` because we already know it's a get.

Reply:
Thanks Tobias for the review! All fixed. I will test it on Linux(already tested)/Mac/Windows(aarch64+x86_64) later. But I don't have ppc and s390 machines so I'm not sure how to test it on them...

Reply:
Looks good. Maybe someone from the OpenJDK community can run some sanity testing on ppc/s390.